packages/utils/typescript/lib/compile.js

```diff
@@ -3,12 +3,12 @@
 const compilers = require('./compilers');
 const getConfigPath = require('./utils/get-config-path');
 
-module.exports = async (srcDir, { watch = false, configOptions = {} } = {}) => {
+module.exports = async (srcDir = "", { watch = false, configOptions = {}, compileOptions = {} } = {}) => {
```

Comment:
Here configOptions was used to store both information for the ts program (`filenames`, etc...) and the compiler (`options`). I would advise renaming this to `options` (which would then contains `compileOptions` instead of `options`). You would then have to rename those variable in the two compilers we define (basic and watch).